app/database/models/user.py

```diff
@@ -116,6 +116,11 @@ def find_by_username(cls, username):
         """Returns the user that has the username we searched for. """
         return cls.query.filter_by(username=username).first()
 
+    @classmethod
+    def find_by_name(cls, name):
+        """Returns the user that has the name we searched for. """
+        return cls.query.filter_by(name=name).first()
```

Comment:
Why first? if I look for Anna I want to find the proper one.

Reply:
The first () will serve the same purpose, user name is unique for all so there would be either just a single search result for a user name if found, else there would be no result.

Reply:
It should return array of users.

Reply:
But what would be the benefit of returning an array? Anyway we would be getting only one user as a response since the display names are unique..

Reply:
Alright, so do you want to say that I should return all users whose name 'contains' the search param rather than name being ' equal' to the search query?

Reply:
Think about Facebook when you type name does it display single user or multiple?

Reply:
One more think it should not be equal it should contain the search part.